unfetter-discover/unfetter-ui
# Issue 994 create

supports unfetter-discover/unfetter#994

This PR will fail until the stix PR is approved  and merged https://github.com/unfetter-discover/stix/pull/12

## problem
* need to add assessment v3 basline selection to assessments v3 (beta) wizard
* the v2 and v3 models were tightly coupled

_Note:_ there might be an issue with dependency injection.  Where by, if you load v3 full-results.effects store and then visit the v2 page, the v2 will use the v3 store.  Things are working and I did not fix this issue

## changes
* decouple models, use the stix package, update stix package with necessary changes
* improved the store error handling, ie added catch inside any effect switchMap and after the map to actions
```
         .switchMap((rollupId: string) => {
             return this.assessService
                 .getByRollupId(rollupId)
                 .mergeMap((data: Assessment[]) => [new SetAssessments(data), new FinishedLoading(true)])
                 .catch((err) => {
                    console.log(err);
                    return Observable.empty();
                });
        });
```
* added deprecation lint WARNING messages to old v2 assessment models

## test
* verify travis
* run the app, verify the functions work in v2 and v3 beta assessments
